src/__tests__/request_middleware.test.ts

```diff
@@ -12,10 +12,20 @@ import {
 
 import { left, right } from "fp-ts/lib/Either";
 
-const ResolvingMiddleware: IRequestMiddleware<never, string> = req => {
+const ResolvingMiddleware: IRequestMiddleware<
+  "IResponseErrorNever",
+  string
+> = req => {
   return Promise.resolve(right<never, string>(req.params.dummy));
 };
 
+const ResolvingNumberMiddleware: IRequestMiddleware<
+  "IResponseErrorNever",
+  number
+> = req => {
+  return Promise.resolve(right<never, number>(1));
+};
+
 const RejectingMiddleware: IRequestMiddleware<
```

Comment:
Could you add a test with a throwing middleware?
As expected we should have an `IResponseErrorInternal`.
```
const DelayedFailureMiddleware: IRequestMiddleware<
"IResponseErrorValidation",
string
> = _ => {
  return new Promise((resolve, reject) => {
    setInterval(() => reject(new Error("Error")), 500)
  })
};
```

Reply:
Sure, you're right.. added in [a461ca0](https://github.com/pagopa/io-ts-commons/pull/370/commits/a461ca09c2743b53456469fa15de2c5fbc6fde42)